src/main/scala/edg_ide/runner/PDFGeneratorUtil.scala

```diff
@@ -1,78 +1,185 @@
 
-      def printNextHierarchyLevel(block: HierarchyBlock, path: DesignPath = DesignPath()): Unit = {
-        val node = HierarchyGraphElk.HBlockToElkNode(block, path, mappers=mappers)
-        printNode(node)
+      // Sets the header and footer of each page. Target of the component target link is set as the header.
+      val targetAnchor = new Anchor(path.toString, font_bold)
+      targetAnchor.setName(path.toString)
+      val headerPhrase = new Phrase
+      headerPhrase.add(targetAnchor)
+      val pageHeader = new HeaderFooter(headerPhrase,false)
+      pageHeader.setAlignment(Element.ALIGN_LEFT)
+      document.setHeader(pageHeader)
 
-        block.blocks.asPairs.map {
-          case (name, subblock) => (name, subblock.`type`)
-        }.collect {
-          case (name, BlockLike.Type.Hierarchy(subblock)) if subblock.blocks.nonEmpty => (path + name, subblock)
-        }.foreach {
-          case (path, subblock) => printNextHierarchyLevel(subblock, path)
+      val pageFooter = new HeaderFooter(new Phrase("Page "), true)
+      pageFooter.setAlignment(Element.ALIGN_RIGHT)
+      pageFooter.setBorder(Rectangle.NO_BORDER)
+      document.setFooter(pageFooter)
+
+      /*
+      Metadata for the Footer does not align the page number correctly if
+      document.open() is called before document.setPageSize() was executed
+       */
+      if (!document.isOpen) {
+        document.open()
+      } else {
+        document.newPage
+      }
+
+      val cb = writer.getDirectContent
+      val graphics = cb.createGraphics(width, adjustedHeight)
+      val painter = new ElkNodePainter(node)
+      painter.paintComponent(graphics, Color.white)
+      graphics.dispose()
+
+      // Prints out table if there are multiple usages of the same component
+      if(dupSet.size > 1) {
+        val table = new PdfPTable(1)
+
+        val message = new Paragraph(s"${className.toSimpleString} is also used at the following:")
+        val title = new PdfPCell(message)
+        title.setBorder(Rectangle.NO_BORDER)
+        table.addCell(title)
+
+        dupSet.foreach { dupPath =>
+          val cellContent = new Paragraph
+          val (parent, current) = dupPath.split
+          if (parent.toString == "(root)") {
+            val parentAnchor = new Anchor
+            parentAnchor.setReference(s"#${parent.toString}")
+            parentAnchor.add(new Phrase(s"${parent.toString}", font))
+            parentAnchor.add(new Phrase(s".${current}"))
+            cellContent.add(parentAnchor)
+          } else {
+            var prevStep = ""
+            dupPath.toString.split('.').foreach { step =>
+              val anchor = new Anchor
+              if (prevStep == "") {
+                anchor.setReference(s"#${step}")
+                anchor.add(new Phrase(s"${step}", font))
+                prevStep = step
+              } else {
+                if (step == dupPath.toString.split('.').last) {
+                  anchor.add(new Phrase(s".${step}"))
```

Comment:
Can you unify the dot handling code? (for both the blue-link-following and last-level-name cases)